File: Python/binary_search/binary_search.py
def binary_search(list, n):
    low = 0;
    high = len(list);
    ret = 0;
    while low <= high:
        guess = int((low + high)/2);
        if list[guess] == n:
            ret = guess;
            break;
        elif list[guess] > n:
            high = guess - 1;
        else:
            low = guess + 1;
    return ret;

print(binary_search([1,2,3,4,5,6],5))

File: JavaScript/binary_search/binary_search.js
function binary_search(list,n) {
  let low = 0;
  let high = list.length;
  let ret = -1;
  while (low <= high) {
    let guess = Math.round((high + low)/2)
    if (list[guess] == n) {
      ret = guess;
      break;
    } else if (list[guess] < n){
      low = guess + 1;
    } else {
      high = guess - 1;
    }
  }
  return ret;
}

console.log(binary_search([1,2,3,4,5,6],5))
